impeller/renderer/backend/vulkan/swapchain_impl_vk.cc

```diff
@@ -377,8 +378,33 @@ bool SwapchainImplVK::Present(const std::shared_ptr<SwapchainImageVK>& image,
   }
 
   const auto& context = ContextVK::Cast(*context_strong);
+  auto current_frame = current_frame_;
+  const auto& sync = synchronizers_[current_frame];
 
-  const auto& sync = synchronizers_[current_frame_];
+  // Submit all command buffers.
+  {
+    auto [fence_result, fence] = context.GetDevice().createFenceUnique({});
+    if (fence_result != vk::Result::eSuccess) {
+      return false;
+    }
+
+    const auto& encoders = context.GetCommandBufferQueue()->Take();
+    vk::SubmitInfo submit_info;
+    std::vector<vk::CommandBuffer> buffers;
+    for (const auto& encoder : encoders) {
+      buffers.push_back(encoder->GetCommandBuffer());
+    }
+    submit_info.setCommandBuffers(buffers);
+    if (context.GetGraphicsQueue()->Submit(submit_info, *fence) !=
+        vk::Result::eSuccess) {
+      return false;
+    }
+
+    if (!context.GetFenceWaiter()->AddFence(
```

Comment:
I think there must be a way to also join this fence/submit with the one below, but I didnt figure it out.